diff --git a/compatibility.h b/compatibility.h
index bdf4b62..1f7e366 100644
--- a/compatibility.h
+++ b/compatibility.h
@@ -78,12 +78,34 @@ static const char *show_deprecated_ ##deprecated_name(void) \
 	return value; \
 }
 
+/*
+ * PostgreSQL version 14+
+ */
+#if PG_VERSION_NUM >= 140000
+
+static void PU_hook(PlannedStmt *pstmt, const char *queryString, bool readOnlyTree,
+					ProcessUtilityContext context, ParamListInfo params,
+					QueryEnvironment *queryEnv,
+					DestReceiver *dest, QueryCompletion *qc);
+#define _PU_HOOK \
+	static void PU_hook(PlannedStmt *pstmt, const char *queryString, bool readOnlyTree,\
+						ProcessUtilityContext context, ParamListInfo params, \
+						QueryEnvironment *queryEnv, \
+						DestReceiver *dest, QueryCompletion *qc)
+
+#define _prev_hook \
+	prev_hook(pstmt, queryString, readOnlyTree, context, params, queryEnv, dest, qc)
+
+#define _standard_ProcessUtility \
+	standard_ProcessUtility(pstmt, queryString, readOnlyTree, context, params, queryEnv, dest, qc)
+
+
 /*
  * PostgreSQL version 13+
  *
  * Introduces QueryCompletion struct
  */
-#if PG_VERSION_NUM >= 130000
+#elif PG_VERSION_NUM >= 130000
 
 static void PU_hook(PlannedStmt *pstmt, const char *queryString,
 					ProcessUtilityContext context, ParamListInfo params,
