diff --git a/set_user.c b/set_user.c
index c1913aa..54b9e3a 100644
--- a/set_user.c
+++ b/set_user.c
@@ -393,12 +393,16 @@ set_user(PG_FUNCTION_ARGS)
 		if (NewUser_is_superuser && Block_LS)
 		{
 			const char	   *old_log_prefix = GetConfigOption("log_line_prefix", true, false);
-			char		   *new_log_prefix = NULL;
+			char		   *new_log_prefix = palloc(strlen(SU_AuditTag) + (old_log_prefix ? strlen(old_log_prefix) + 3 : 1));
 
 			if (old_log_prefix)
-				new_log_prefix = psprintf("%s%s: ", old_log_prefix, SU_AuditTag);
+			{
+				strcpy(new_log_prefix, old_log_prefix);
+				strcat(new_log_prefix, SU_AuditTag);
+				strcat(new_log_prefix, ": ");
+			}
 			else
-				new_log_prefix = pstrdup(SU_AuditTag);
+				strcpy(new_log_prefix, SU_AuditTag);
 
 			/*
 			 * Force logging of everything if block_log_statement is true
