--- debian/rules.orig	2018-01-30 06:06:47.000000000 +0000
+++ debian/rules	2018-04-28 11:31:55.013154357 +0000
@@ -101,38 +101,43 @@
 			--with-rename-internal-libtiff-symbols=yes \
 			--with-rename-internal-libgeotiff-symbols=yes \
 			--with-threads \
-			--with-armadillo=yes \
+			--with-armadillo=no \
 			--with-cfitsio=no \
-			--with-curl \
+			--with-curl=no \
 			--with-ecw=no \
-			--with-epsilon=/usr \
-			--with-freexl=yes \
+			--with-epsilon=no \
+			--with-freexl=no \
 			--with-geos \
-			--with-geotiff=yes \
+			--with-geotiff=internal \
 			--with-grass=no \
-			$(WITH_HDF5) \
 			--with-libjson-c=/usr \
-			--with-libkml=yes \
+			--with-libkml=no \
 			--with-liblzma=yes \
-			--with-libtiff=yes \
+			--with-libtiff=internal \
 			--with-mrsid=no \
-			--with-mysql \
-			--with-netcdf \
-			--with-odbc \
-			--with-ogdi \
-			--with-openjpeg=yes \
-			--with-pg \
-			--with-poppler=yes \
-			--with-qhull=yes \
-			--with-sosi=yes \
-			--with-spatialite=/usr \
-			--with-sqlite3 \
-			--with-webp \
-			--with-xerces \
-			--with-dods-root=/usr \
+			--without-mysql \
+			--without-netcdf \
+			--with-odbc=no \
+			--without-ogdi \
+			--without-jpeg12 \
+			--with-jpeg=internal \
+			--with-png=internal \
+			--with-openjpeg=no \
+			--with-gif=no \
+			--with-hdf4=no \
+			--with-hdf5=no \
+			--without-pg \
+			--with-poppler=no \
+			--with-qhull=no \
+			--with-sosi=no \
+			--with-spatialite=no \
+			--with-sqlite3=no \
+			--with-webp=no \
+			--without-xerces \
+			--with-dods-root=no \
 			--with-static-proj4=yes \
-			--with-java=/usr/lib/jvm/default-java \
-			--with-perl \
+			--without-java \
+			--without-perl \
 			--with-python; \
 		mv GDALmake.opt GDALmake.opt-$$V; \
 	done
