--- debian/rules.orig	2022-08-30 11:19:02.338976183 +0000
+++ debian/rules	2022-08-30 11:32:40.334594832 +0000
@@ -130,42 +130,41 @@
 			--with-ssse3=$(WITH_SSSE3) \
 			--with-avx=$(WITH_AVX) \
 			--with-bash-completion=yes \
-			--with-armadillo=yes \
+			--with-armadillo=no \
 			--with-blosc=yes \
 			--with-cfitsio=yes \
 			--with-charls \
-			--with-curl \
+			--with-curl=no \
 			--with-libdeflate \
 			--with-ecw=no \
-			--with-freexl=yes \
+			--with-freexl=no \
 			--with-geos \
-			--with-geotiff=yes \
+			--with-geotiff=internal \
 			--with-grass=no \
-			$(WITH_HDF5) \
 			--with-heif \
 			--with-libjson-c=/usr \
-			--with-libkml=yes \
+			--with-libkml=no \
 			--with-liblzma=yes \
-			--with-libtiff=yes \
+			--with-libtiff=internal \
 			--with-lz4=yes \
 			--with-mrsid=no \
-			--with-mysql \
-			--with-netcdf \
-			--with-odbc \
-			--with-ogdi \
-			--with-openjpeg=yes \
-			--with-pcre2 \
-			--with-pg \
-			--with-poppler=yes \
-			--with-proj=yes \
-			--with-qhull=yes \
-			--with-sosi=yes \
-			--with-spatialite=/usr \
-			--with-sqlite3 \
-			--with-webp \
-			--with-xerces \
+			--without-mysql \
+			--without-netcdf \
+			--with-odbc=no \
+			--without-ogdi \
+			--with-openjpeg=no \
+			--with-pcre2=no \
+			--without-pg \
+			--with-poppler=no \
+			--with-proj \
+			--with-qhull=no \
+			--with-sosi=no \
+			--with-spatialite=no \
+			--with-sqlite3=no \
+			--with-webp=no \
+			--with-xerces=no \
 			--with-zstd \
-			--with-perl \
+			--without-perl \
 			--with-python; \
 		mv GDALmake.opt GDALmake.opt-$$V; \
 	done
