diff -uN debian.orig/rules debian/rules
--- debian.orig/rules	2017-11-15 13:20:10.667571954 +0000
+++ debian/rules	2017-11-15 13:19:54.783631276 +0000
@@ -21,8 +21,8 @@
 		-DCMAKE_INSTALL_PREFIX=/usr \
 		-DCMAKE_INSTALL_LIBDIR=lib/$(DEB_HOST_MULTIARCH) \
 		-DCMAKE_VERBOSE_MAKEFILE=ON \
-		-DSFCGAL_BUILD_TESTS=ON \
-		-DSFCGAL_BUILD_VIEWER=ON
+		-DSFCGAL_BUILD_TESTS=OFF \
+		-DSFCGAL_BUILD_VIEWER=OFF
 
 override_dh_install:	
 	# removing embedded rpath in binaries
diff -uN debian.orig/sfcgal-bin.install debian/sfcgal-bin.install
--- debian.orig/sfcgal-bin.install	2015-11-16 18:33:07.000000000 +0000
+++ debian/sfcgal-bin.install	2017-11-15 13:20:19.931537328 +0000
@@ -1 +0,0 @@
-usr/bin/viewer-SFCGAL
