diff -urN pg-partman-3.1.3.orig/src/pg_partman_bgw.c pg-partman-3.1.3/src/pg_partman_bgw.c
--- pg-partman-3.1.3.orig/src/pg_partman_bgw.c	2018-04-16 21:27:46.000000000 +0200
+++ pg-partman-3.1.3/src/pg_partman_bgw.c	2018-05-27 17:19:05.717459837 +0200
@@ -395,7 +395,11 @@
     }
 
     elog(DEBUG1, "Before run_maint initialize connection for db %s", dbname);
+#if PG_VERSION_NUM >= 110000
+    BackgroundWorkerInitializeConnection(dbname, pg_partman_bgw_role, 0);
+#else
     BackgroundWorkerInitializeConnection(dbname, pg_partman_bgw_role);
+#endif
     elog(DEBUG1, "After run_maint initialize connection for db %s", dbname);
 
     initStringInfo(&buf);
