diff -urN pgextwlist-1.6.orig/pgextwlist.c pgextwlist-1.6/pgextwlist.c
--- pgextwlist-1.6.orig/pgextwlist.c	2017-10-15 14:39:40.000000000 +0200
+++ pgextwlist-1.6/pgextwlist.c	2018-05-27 17:22:57.506410963 +0200
@@ -132,56 +132,30 @@
 void
 _PG_init(void)
 {
-  PG_TRY();
-  {
-    extwlist_extensions = GetConfigOptionByName("extwlist.extensions", NULL
-#if PG_MAJOR_VERSION >= 906
-												, false
-#endif
-												);
-  }
-  PG_CATCH();
-  {
-	  DefineCustomStringVariable("extwlist.extensions",
-								 "List of extensions that are whitelisted",
-								 "Separated by comma",
-								 &extwlist_extensions,
-								 "",
-								 PGC_SUSET,
-								 GUC_NOT_IN_SAMPLE,
-								 NULL,
-								 NULL,
-								 NULL);
-    EmitWarningsOnPlaceholders("extwlist.extensions");
-  }
-  PG_END_TRY();
+	DefineCustomStringVariable("extwlist.extensions",
+								"List of extensions that are whitelisted",
+								"Separated by comma",
+								&extwlist_extensions,
+								"",
+								PGC_SUSET,
+								GUC_NOT_IN_SAMPLE,
+								NULL,
+								NULL,
+								NULL);
 
-  PG_TRY();
-  {
-    extwlist_custom_path = GetConfigOptionByName("extwlist.custom_path", NULL
-#if PG_MAJOR_VERSION >= 906
-												, false
-#endif
-												);
-  }
-  PG_CATCH();
-  {
-	  DefineCustomStringVariable("extwlist.custom_path",
-								 "Directory where to load custom scripts from",
-								 "",
-								 &extwlist_custom_path,
-								 "",
-								 PGC_SUSET,
-								 GUC_NOT_IN_SAMPLE,
-								 NULL,
-								 NULL,
-								 NULL);
-    EmitWarningsOnPlaceholders("extwlist.custom_path");
-  }
-  PG_END_TRY();
+	DefineCustomStringVariable("extwlist.custom_path",
+								"Directory where to load custom scripts from",
+								"",
+								&extwlist_custom_path,
+								"",
+								PGC_SUSET,
+								GUC_NOT_IN_SAMPLE,
+								NULL,
+								NULL,
+								NULL);
 
-  prev_ProcessUtility = ProcessUtility_hook;
-  ProcessUtility_hook = extwlist_ProcessUtility;
+	prev_ProcessUtility = ProcessUtility_hook;
+	ProcessUtility_hook = extwlist_ProcessUtility;
 }
 
 /*
