diff -urN pgextwlist-1.11.orig/pgextwlist.c pgextwlist-1.11/pgextwlist.c
--- pgextwlist-1.11.orig/pgextwlist.c	2020-05-24 21:10:12.000000000 +0200
+++ pgextwlist-1.11/pgextwlist.c	2021-06-25 13:16:13.568885524 +0200
@@ -101,7 +101,7 @@
 
 #define PROCESS_UTILITY_ARGS pstmt, queryString, context, \
                               params, queryEnv, dest, completionTag
-#else
+#elif PG_MAJOR_VERSION < 1400
 #define PROCESS_UTILITY_PROTO_ARGS PlannedStmt *pstmt,                    \
 										const char *queryString,       \
 										ProcessUtilityContext context, \
@@ -112,6 +112,18 @@
 
 #define PROCESS_UTILITY_ARGS pstmt, queryString, context, \
                               params, queryEnv, dest, qc
+#else
+#define PROCESS_UTILITY_PROTO_ARGS PlannedStmt *pstmt,                    \
+										const char *queryString,       \
+										bool readOnlyTree, \
+										ProcessUtilityContext context, \
+										ParamListInfo params,          \
+										QueryEnvironment *queryEnv,    \
+										DestReceiver *dest,            \
+										QueryCompletion *qc
+#define PROCESS_UTILITY_ARGS pstmt, queryString, readOnlyTree, context, \
+                              params, queryEnv, dest, qc
+
 #endif	/* PG_MAJOR_VERSION */
 
 #define EREPORT_EXTENSION_IS_NOT_WHITELISTED(op)						\
diff -urN pgextwlist-1.11.orig/utils.c pgextwlist-1.11/utils.c
--- pgextwlist-1.11.orig/utils.c	2020-05-24 21:10:12.000000000 +0200
+++ pgextwlist-1.11/utils.c	2021-06-25 13:16:20.133002157 +0200
@@ -470,6 +470,9 @@
 			{
 				ProcessUtility(stmt,
 							   sql,
+#if PG_MAJOR_VERSION >= 1400
+							   false,
+#endif
 #if PG_MAJOR_VERSION >= 903
 							   PROCESS_UTILITY_QUERY,
 #endif
