diff -urN pgq-3.2.6-34-g50068b2.backup/lowlevel/insert_event.c pgq-3.2.6-34-g50068b2/lowlevel/insert_event.c
--- pgq-3.2.6-34-g50068b2.backup/lowlevel/insert_event.c	2019-09-13 06:50:36.000000000 +0000
+++ pgq-3.2.6-34-g50068b2/lowlevel/insert_event.c	2020-05-21 18:46:57.170486231 +0000
@@ -26,6 +26,7 @@
 #include "utils/builtins.h"
 #include "utils/datetime.h"
 #include "utils/hsearch.h"
+#include "access/hash.h"
 #include "access/xact.h"
 
 /*
diff -urN pgq-3.2.6-34-g50068b2.backup/triggers/common.c pgq-3.2.6-34-g50068b2/triggers/common.c
--- pgq-3.2.6-34-g50068b2.backup/triggers/common.c	2019-09-13 06:50:36.000000000 +0000
+++ pgq-3.2.6-34-g50068b2/triggers/common.c	2020-05-21 18:46:32.061484634 +0000
@@ -18,6 +18,7 @@
 
 #include <postgres.h>
 
+#include <access/hash.h>
 #include <commands/trigger.h>
 #include <catalog/pg_type.h>
 #include <catalog/pg_namespace.h>
