diff -urN postgresql-plproxy-2.8.orig/src/cluster.c postgresql-plproxy-2.8/src/cluster.c
--- postgresql-plproxy-2.8.orig/src/cluster.c	2017-10-08 10:53:35.000000000 +0200
+++ postgresql-plproxy-2.8/src/cluster.c	2018-05-27 17:00:56.482207603 +0200
@@ -162,7 +162,12 @@
 	 * create long-lived memory context
 	 */
 
-	cluster_mem = AllocSetContextCreate(TopMemoryContext,
+#if PG_VERSION_NUM >= 110000
+	cluster_mem = AllocSetContextCreateExtended
+#else
+	cluster_mem = AllocSetContextCreate
+#endif
+									   (TopMemoryContext,
 										"PL/Proxy cluster context",
 										ALLOCSET_SMALL_MINSIZE,
 										ALLOCSET_SMALL_INITSIZE,
@@ -588,7 +593,11 @@
 	 */
 	aclresult = pg_foreign_server_aclcheck(um->serverid, um->userid, ACL_USAGE);
 	if (aclresult != ACLCHECK_OK)
+#if PG_VERSION_NUM >= 110000
+		aclcheck_error(aclresult, OBJECT_FOREIGN_SERVER, cluster->name);
+#else
 		aclcheck_error(aclresult, ACL_KIND_FOREIGN_SERVER, cluster->name);
+#endif
 
 	/* Extract the common connect string elements from user mapping */
 	got_user = false;
@@ -657,7 +666,11 @@
 	 */
 	aclresult = pg_foreign_server_aclcheck(foreign_server->serverid, info->user_oid, ACL_USAGE);
 	if (aclresult != ACLCHECK_OK)
+#if PG_VERSION_NUM >= 110000
+		aclcheck_error(aclresult, OBJECT_FOREIGN_SERVER, foreign_server->servername);
+#else
 		aclcheck_error(aclresult, ACL_KIND_FOREIGN_SERVER, foreign_server->servername);
+#endif
 
 	/* drop old config values */
 	clear_config(&cluster->config);
diff -urN postgresql-plproxy-2.8.orig/src/function.c postgresql-plproxy-2.8/src/function.c
--- postgresql-plproxy-2.8.orig/src/function.c	2017-10-08 10:53:35.000000000 +0200
+++ postgresql-plproxy-2.8/src/function.c	2018-05-27 17:06:33.455411217 +0200
@@ -214,8 +214,13 @@
 	Form_pg_proc proc_struct;
 	proc_struct = (Form_pg_proc) GETSTRUCT(proc_tuple);
 	if (proc_struct->prorettype == RECORDOID
+#if PG_VERSION_NUM >= 110000
+		&& (heap_attisnull(proc_tuple, Anum_pg_proc_proargmodes, NULL)
+		    || heap_attisnull(proc_tuple, Anum_pg_proc_proargnames, NULL)))
+#else
 		&& (heap_attisnull(proc_tuple, Anum_pg_proc_proargmodes)
 		    || heap_attisnull(proc_tuple, Anum_pg_proc_proargnames)))
+#endif
 		return true;
 	return false;
 }
@@ -233,7 +238,12 @@
 	MemoryContext f_ctx,
 				old_ctx;
 
-	f_ctx = AllocSetContextCreate(TopMemoryContext,
+#if PG_VERSION_NUM >= 110000
+	f_ctx = AllocSetContextCreateExtended
+#else
+	f_ctx = AllocSetContextCreate
+#endif
+								 (TopMemoryContext,
 								  "PL/Proxy function context",
 								  ALLOCSET_SMALL_MINSIZE,
 								  ALLOCSET_SMALL_INITSIZE,
