diff --git a/.travis.yml b/.travis.yml
index 47d0101..72b1cf6 100644
--- a/.travis.yml
+++ b/.travis.yml
@@ -4,6 +4,8 @@ dist: xenial
 sudo: required
 
 env:
+  - PGVER=13
+    PGTESTING=1
   - PGVER=12
   - PGVER=11
   - PGVER=10
diff --git a/doc/pg_repack.rst b/doc/pg_repack.rst
index efa60bc..11b014e 100644
--- a/doc/pg_repack.rst
+++ b/doc/pg_repack.rst
@@ -40,7 +40,7 @@ Requirements
 ------------
 
 PostgreSQL versions
-    PostgreSQL 9.1, 9.2, 9.3, 9.4, 9.5, 9.6, 10, 11, 12
+    PostgreSQL 9.1, 9.2, 9.3, 9.4, 9.5, 9.6, 10, 11, 12, 13
 
 Disks
     Performing a full-table repack requires free disk space about twice as
diff --git a/lib/repack.c b/lib/repack.c
index d75f256..89d8073 100644
--- a/lib/repack.c
+++ b/lib/repack.c
@@ -19,7 +19,6 @@
 
 /*
  * heap_open/heap_close was moved to table_open/table_close in 12.0
- * table.h has macros mapping the old names to the new ones
  */
 #if PG_VERSION_NUM >= 120000
 #include "access/table.h"
@@ -1169,7 +1168,11 @@ swap_heap_or_index_files(Oid r1, Oid r2)
 	CatalogIndexState indstate;
 
 	/* We need writable copies of both pg_class tuples. */
+#if PG_VERSION_NUM >= 120000
+	relRelation = table_open(RelationRelationId, RowExclusiveLock);
+#else
 	relRelation = heap_open(RelationRelationId, RowExclusiveLock);
+#endif
 
 	reltup1 = SearchSysCacheCopy(RELOID,
 								 ObjectIdGetDatum(r1),
@@ -1332,7 +1335,11 @@ swap_heap_or_index_files(Oid r1, Oid r2)
 	heap_freetuple(reltup1);
 	heap_freetuple(reltup2);
 
+#if PG_VERSION_NUM >= 120000
+	table_close(relRelation, RowExclusiveLock);
+#else
 	heap_close(relRelation, RowExclusiveLock);
+#endif
 }
 
 /**
