diff --git a/wal2json.c b/wal2json.c
index 75b33c8..c4772e5 100644
--- a/wal2json.c
+++ b/wal2json.c
@@ -820,7 +820,12 @@ pg_decode_begin_txn_v1(LogicalDecodingContext *ctx, ReorderBufferTXN *txn)
 	}
 
 	if (data->include_timestamp)
+#if	PG_VERSION_NUM >= 150000
+		appendStringInfo(ctx->out, "%s\"timestamp\":%s\"%s\",%s", data->ht, data->sp, timestamptz_to_str(txn->xact_time.commit_time), data->nl);
+
+#else
 		appendStringInfo(ctx->out, "%s\"timestamp\":%s\"%s\",%s", data->ht, data->sp, timestamptz_to_str(txn->commit_time), data->nl);
+#endif
 
 #if PG_VERSION_NUM >= 90500
 	if (data->include_origin)
@@ -847,7 +852,11 @@ pg_decode_begin_txn_v2(LogicalDecodingContext *ctx, ReorderBufferTXN *txn)
 	if (data->include_xids)
 		appendStringInfo(ctx->out, ",\"xid\":%u", txn->xid);
 	if (data->include_timestamp)
-			appendStringInfo(ctx->out, ",\"timestamp\":\"%s\"", timestamptz_to_str(txn->commit_time));
+#if	PG_VERSION_NUM >= 150000
+		appendStringInfo(ctx->out, ",\"timestamp\":\"%s\"", timestamptz_to_str(txn->xact_time.commit_time));
+#else
+		appendStringInfo(ctx->out, ",\"timestamp\":\"%s\"", timestamptz_to_str(txn->commit_time));
+#endif
 
 #if PG_VERSION_NUM >= 90500
 	if (data->include_origin)
@@ -876,7 +885,9 @@ pg_decode_commit_txn(LogicalDecodingContext *ctx, ReorderBufferTXN *txn,
 {
 	JsonDecodingData *data = ctx->output_plugin_private;
 
-#if PG_VERSION_NUM >= 100000
+#if	PG_VERSION_NUM >= 150000
+	OutputPluginUpdateProgress(ctx, false);
+#elif PG_VERSION_NUM >= 100000
 	OutputPluginUpdateProgress(ctx);
 #endif
 
@@ -925,7 +936,12 @@ pg_decode_commit_txn_v2(LogicalDecodingContext *ctx, ReorderBufferTXN *txn,
 	if (data->include_xids)
 		appendStringInfo(ctx->out, ",\"xid\":%u", txn->xid);
 	if (data->include_timestamp)
-			appendStringInfo(ctx->out, ",\"timestamp\":\"%s\"", timestamptz_to_str(txn->commit_time));
+
+#if	PG_VERSION_NUM >= 150000
+		appendStringInfo(ctx->out, ",\"timestamp\":\"%s\"", timestamptz_to_str(txn->xact_time.commit_time));
+#else
+		appendStringInfo(ctx->out, ",\"timestamp\":\"%s\"", timestamptz_to_str(txn->commit_time));
+#endif
 
 #if PG_VERSION_NUM >= 90500
 	if (data->include_origin)
@@ -2208,7 +2224,12 @@ pg_decode_write_change(LogicalDecodingContext *ctx, ReorderBufferTXN *txn, Relat
 		appendStringInfo(ctx->out, ",\"xid\":%u", txn->xid);
 
 	if (data->include_timestamp)
+#if	PG_VERSION_NUM >= 150000
+		appendStringInfo(ctx->out, ",\"timestamp\":\"%s\"", timestamptz_to_str(txn->xact_time.commit_time));
+#else
 		appendStringInfo(ctx->out, ",\"timestamp\":\"%s\"", timestamptz_to_str(txn->commit_time));
+#endif
+
 
 #if PG_VERSION_NUM >= 90500
 	if (data->include_origin)
@@ -2520,7 +2541,12 @@ pg_decode_message_v2(LogicalDecodingContext *ctx, ReorderBufferTXN *txn,
 	if (data->include_timestamp)
 	{
 		if (transactional)
+#if	PG_VERSION_NUM >= 150000
+			appendStringInfo(ctx->out, ",\"timestamp\":\"%s\"", timestamptz_to_str(txn->xact_time.commit_time));
+#else
 			appendStringInfo(ctx->out, ",\"timestamp\":\"%s\"", timestamptz_to_str(txn->commit_time));
+#endif
+
 		else
 			appendStringInfoString(ctx->out, ",\"timestamp\":null");
 	}
@@ -2638,7 +2664,12 @@ static void pg_decode_truncate_v1(LogicalDecodingContext *ctx,
 		appendStringInfo(ctx->out, "%s%s%s\"xid\":%s%u,%s", data->ht, data->ht, data->ht, data->sp, txn->xid, data->nl);
 
 	if (data->include_timestamp)
+#if	PG_VERSION_NUM >= 150000
+		appendStringInfo(ctx->out, "%s%s%s\"timestamp\":%s\"%s\",%s", data->ht, data->ht, data->ht, data->sp, timestamptz_to_str(txn->xact_time.commit_time), data->nl);
+#else
 		appendStringInfo(ctx->out, "%s%s%s\"timestamp\":%s\"%s\",%s", data->ht, data->ht, data->ht, data->sp, timestamptz_to_str(txn->commit_time), data->nl);
+#endif
+
 
 	if (data->include_origin)
 		appendStringInfo(ctx->out, "%s%s%s\"origin\":%s%u,%s", data->ht, data->ht, data->ht, data->sp, txn->origin_id, data->nl);
@@ -2723,7 +2754,12 @@ static void pg_decode_truncate_v2(LogicalDecodingContext *ctx,
 			appendStringInfo(ctx->out, ",\"xid\":%u", txn->xid);
 
 		if (data->include_timestamp)
+#if	PG_VERSION_NUM >= 150000
+			appendStringInfo(ctx->out, ",\"timestamp\":\"%s\"", timestamptz_to_str(txn->xact_time.commit_time));
+#else
 			appendStringInfo(ctx->out, ",\"timestamp\":\"%s\"", timestamptz_to_str(txn->commit_time));
+#endif
+
 
 		if (data->include_origin)
 			appendStringInfo(ctx->out, ",\"origin\":%u", txn->origin_id);
